src/bm_sim/P4Objects.cpp

```diff
@@ -2732,6 +2755,23 @@ P4Objects::remove_json_value(const std::string &type, const std::string &name) {
       }
       (*cfg_pipeline_conditionals_map[pipeline_name]) = newArray;
     }
+  } else if (type == "register_array"){
+    if (cfg_register_arrays_map.find(name) == cfg_register_arrays_map.end()) {
+      std::cout << "remove_json_value: no register_array name: " << name << std::endl;
+    } else {
+      Json::Value newArray = Json::arrayValue;
+      for (const auto& cfg_register_array : cfg_root["register_arrays"]) {
+        if (cfg_register_array["name"] != name) {
+          newArray.append(cfg_register_array);
+        }
+      }
+      cfg_root["register_arrays"] = newArray;
```

Comment:
2763-2769, is there a better way to remove the item from cfg_register_array directly? What's the reason for using this way?